packages/hierarchies-react/src/test/UseSelectionHandler.test.tsx

```diff
@@ -0,0 +1,417 @@
+
+import { expect } from "chai";
+import sinon from "sinon";
+import { UserEvent } from "@testing-library/user-event";
+import { PresentationHierarchyNode, PresentationInfoNode, PresentationTreeNode } from "../presentation-hierarchies-react/Types";
+import { SelectionChangeType, SelectionMode, useSelectionHandler } from "../presentation-hierarchies-react/UseSelectionHandler";
+import { render } from "./TestUtils";
+
+interface TestComponentProps {
+  rootNodes: Array<PresentationTreeNode> | undefined;
+  selectNodes: (nodeIds: Array<string>, changeType: SelectionChangeType) => void;
+  selectionMode: SelectionMode;
+  isSelected: boolean;
+}
+
+function TestComponent({ rootNodes, selectNodes, selectionMode, isSelected }: TestComponentProps) {
+  const { onNodeKeyDown, onNodeClick } = useSelectionHandler({ rootNodes, selectNodes, selectionMode });
+  const invalidNode = { id: "invalid" } as PresentationHierarchyNode;
+  const nodes = rootNodes ? [...rootNodes, invalidNode] : [invalidNode];
+  return (
+    <>
+      {nodes?.map((node) => {
+        return (
+          <div
+            role="button"
+            key={node.id}
+            tabIndex={0}
+            onClick={(e) => onNodeClick(node.id, isSelected, e)}
+            onKeyDown={(e) => onNodeKeyDown(node.id, isSelected, e)}
+          >
+            {node.id}
+          </div>
+        );
+      })}
+    </>
+  );
+}
+
+describe("useSelectionHandler", () => {
+  const selectNodesStub = sinon.stub<[Array<string>, SelectionChangeType], void>();
+
+  const createHierarchyNode = (id: string, children: Array<PresentationTreeNode> = [], isExpanded: boolean = true) => {
+    return { id, isExpanded, children } as PresentationHierarchyNode;
+  };
+
+  const createInfoNode = (id: string) => {
+    return { id, message: "message" } as PresentationInfoNode;
+  };
+
+  const createProps = (rootNodes: Array<PresentationTreeNode> | undefined, selectionMode: SelectionMode, isSelected: boolean) => {
+    return { rootNodes, selectNodes: selectNodesStub, selectionMode, isSelected };
+  };
+
+  afterEach(() => {
+    selectNodesStub.reset();
+  });
+
+  const selectionTests = (clickNode: (user: UserEvent, node: HTMLElement) => Promise<void>) => {
+    it("does nothing when no root nodes passed", async () => {
+      const { user, getByText } = render(<TestComponent {...createProps(undefined, "none", true)} />);
+
+      const node = getByText("invalid");
+      node.focus();
+      await clickNode(user, node);
+
+      expect(selectNodesStub).to.not.be.called;
+    });
+
+    describe("`none` selection mode", () => {
+      it("does nothing when node is clicked", async () => {
+        const rootNode = createHierarchyNode("node");
+        const { user, getByText } = render(<TestComponent {...createProps([rootNode], "none", true)} />);
+
+        const node = getByText("node");
+        node.focus();
+
+        await user.keyboard(`{Control>}`);
```

Comment:
Why `Ctrl` used here?

Reply:
Added these extra keys to check if `ctrl` and `shift` would be ignored in non extended selection mode. Moved this into separate tests to be more clear.